uat/test_uat_publish.py

```diff
@@ -0,0 +1,96 @@
+import json
+import os
+import subprocess as sp
+from pathlib import Path
+
+import t_utils
+
+
+def test_publish_template_zip(change_test_dir):
+    # Recipe contains HelloWorld.zip artifact. So, create HelloWorld directory inside temporary directory.
+    path_HelloWorld = Path(change_test_dir).joinpath("HelloWorld")
+    component_name = "com.example.PythonHelloWorld"
+    region = "us-east-1"
+    # Check if init downloads templates with necessary files.
+    check_init_template = sp.run(
+        ["gdk", "component", "init", "-t", "HelloWorld", "-l", "python", "-n", "HelloWorld"], check=True, stdout=sp.PIPE
+    )
+    assert check_init_template.returncode == 0
+    assert Path(path_HelloWorld).joinpath("recipe.yaml").resolve().exists()
+    config_file = Path(path_HelloWorld).joinpath("gdk-config.json").resolve()
+    assert config_file.exists()
+
+    # Update gdk-config file mandatory field like region.
+    with open(str(config_file), "r") as f:
+        config = json.loads(f.read())
+        config["component"][component_name]["author"] = "gdk-cli-uat"
+        config["component"][component_name]["publish"]["region"] = region
+        config["component"][component_name]["publish"]["bucket"] = "gdk-cli-uat"
+    with open(str(config_file), "w") as f:
+        f.write(json.dumps(config))
```

Comment:
can we take this  out as method to reuse?

Reply:
Done